src/snowflake/snowpark/modin/plugin/extensions/index.py

```diff
@@ -1392,8 +1395,27 @@ def nunique(self) -> None:
         --------
         DataFrame.nunique: Method nunique for DataFrame.
         Series.count: Count non-NA/null observations in the Series.
-        """
-        # TODO: SNOW-1458132 implement nunique
+
+        Examples
+        --------
+        >>> s = pd.Series([1, 3, 5, 7, 7])
+        >>> s
+        0    1
+        1    3
+        2    5
+        3    7
+        4    7
+        dtype: int64
+
+        >>> s.nunique()
+        4
+        """
+        return (
+            self._query_compiler.reset_index()
+            .nunique(axis=0, dropna=dropna)
+            .to_pandas()
+            .iloc[0, 0]
+        )
```

Comment:
@sfc-gh-yzou is correct that we should also avoid `reset_index()` here to reduce the number of `row_number` calls. We can do this, except we should probably abstract the `modin_frame` method calls into a separate method in the query compiler:


```suggestion
        from snowflake.snowpark.functions import col        
        return (
            SnowflakeQueryCompiler(self._query_compiler._modin_frame.append_column('index', col(self._query_compiler._modin_frame.index_column_snowflake_quoted_identifiers[0])))
            .nunique(axis=0, dropna=dropna)
            .to_pandas()
            .iloc[0, 0]
        )
```